cmd/opera/launcher/validator.go

```diff
@@ -33,38 +33,28 @@ var validatorPasswordFlag = cli.StringFlag{
 // command line flags or from the keystore if CLI indexed.
 func setValidator(ctx *cli.Context, cfg *emitter.Config) error {
 	// Extract the current validator address, new flag overriding legacy one
-	var validatorID idx.ValidatorID
-	var validatorPubkey validatorpk.PubKey
 	var err error
 	if ctx.GlobalIsSet(FakeNetFlag.Name) {
```

Comment:
Just curious what will happen if both `ctx.GlobalIsSet(FakeNetFlag.Name)` and `ctx.GlobalIsSet(validatorIDFlag.Name)` evaluate to `true` ? First case one updates `cfg.Validator.ID, num, err = parseFakeGen(ctx.GlobalString(FakeNetFlag.Name))`, the second one would override  `cfg.Validator.ID`  with `idx.ValidatorID(ctx.GlobalInt(validatorIDFlag.Name))`. Just wanna make sure there is no conflict here.

Reply:
If both `--fakenet` and `--validator.id` are set, then `--validator.id` will overrule fakenet's validator ID. The same with `--validator.pubkey`
There are situations when you need to use both `--fakenet` and `--validator.id`/`--validator.pubkey`, e.g.:
- when you staked a new validator on fakenet and need to set up a validator for it. You would need to write something like `--fakenet 0/3 --validator.id 4 --validator.pubkey 0xsomehex`. Yet, I don't think there are any situations when you have to use `--fakenet` with non-zero first argument (e.g. `--fakenet 1/3`) and --validator.id. So we may restrict it, e.g.:
```
	if ctx.GlobalIsSet(validatorIDFlag.Name) && ctx.GlobalIsSet(FakeNetFlag.Name) && cfg.Validator.ID != 0 {
		return errors.New("specified validator ID with both --fakenet and --validator.id")
	}
	if ctx.GlobalIsSet(validatorIDFlag.Name) {
```
Do you think we should add such restriction?

Reply:
yeah we have to restrict this unexpected behavior `--fakenet 0/3 --validator.id 4 --validator.pubkey 0xsomehex` if there are no situations where we can set all three flags at once. I can open a separate PR for it. Does it sound good to you?

Reply:
You misunderstood - there are some cases when using all 3 flags will make sense, and other cases when it has no sense
In your example, it may make sense. User may use `--fakenet 0/3` just to specify the network, and then `--validator.id 4 --validator.pubkey 0xsomehex` to specify validator. Normally, when you're working in fakenet, there's a shortcut for that - you can just use `--fakenet 1/3` if you also want to run a validator. However, fakenet is an unrestricted network, and you may create new validators at will. This shortcut (`--fakenet 1/3` ) works only for genesis fakenet validators, for a newly created validator user has to use the `--validator.id 4 --validator.pubkey 0xsomehex`

Reply:
```
	if ctx.GlobalIsSet(validatorIDFlag.Name) && ctx.GlobalIsSet(FakeNetFlag.Name) && cfg.Validator.ID != 0 {
		return errors.New("specified validator ID with both --fakenet and --validator.id")
	}
	if ctx.GlobalIsSet(validatorIDFlag.Name) {
```
In other words, if user specified `fakenet` with some non-zero validator, and then proceeded with overriding this validator with `--validator.id`, then it doesn't have a lot of sense (at least to me). It would be clearer if user specified `fakenet` with zero validator in this case

Reply:
Thanks for clarification. Yeah, I lack some context here. 
I assume, this case makes no sense. That's why you yield an error here.
```
if ctx.GlobalIsSet(validatorIDFlag.Name) && ctx.GlobalIsSet(FakeNetFlag.Name) && cfg.Validator.ID != 0 {
	return errors.New("specified validator ID with both --fakenet and --validator.id")
}
```
You mentioned there are more cases where it does not make sense. Would you like to add them as well?

Reply:
I believe this `if block` addresses all "silly" cases